pkg/metrics/types.go

```diff
@@ -77,12 +77,13 @@ const (
 
 // Metric defines a unique metric with metadata.
 type Metric struct {
-	MetricSource  MetricSource
-	MetricType    MetricType
-	PromQL        string // Optional: Only applicable for PromQL-based metrics
-	RawMetricName string // Optional: Only applicable for QueryLabel-based metrics
-	Description   string
-	MetricScope   MetricScope
+	MetricSource         MetricSource
+	MetricType           MetricType
+	PromQL               string            // Optional: Only applicable for PromQL-based metrics
+	RawMetricName        string            // Optional: Only applicable for QueryLabel-based metrics
+	RawMetricNameMapping map[string]string // Optional: Mapping from engine type to raw metric name.
```

Comment:
with the mapping, the rawMetricsName should be logical metrics names? We need to consider the compatibility as well